### ISSUE TITLE
Django form fields not well supported

### ISSUE THREAD

**Issue by nimasmi** [torchbox/django-pattern-library#113]
If a template contains `{% for field in form %}` or even `{% if form %}`, then it's easy enough to render in django-pattern-library so long as we force the form to be null in the YAML context, and are happy not to have the form.

If the form is rendered explicitly by field names, then it requires a lot more work, and support for this is not documented. I had a quick go, but my template has things like:

```htmldjango
            {% include "patterns/molecules/form_field.html" with field=form.email %}
```

and that template has a custom filter: `{{ field|field_type }}` which necessitated YAML like:

```yaml
  form:
    email:
      bound_field:
        field:
          widget:
            class:
              __name__: char_field
```

I didn't go much further, because these is even more convoluted markup in my chosen template, and it wasn't a high enough priority.

**Comment by thibaudcolas** [comment]
Related: #106.

**Comment by bcdickinson** [comment]
Also related: #114

**Comment by thibaudcolas** [comment]
Mentioned in the docs as of #131.

**Comment by thibaudcolas** [comment]
Fixed in #141 :)
